internal/tui/tui.go

```diff
@@ -69,19 +60,19 @@ func (m *model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 	// Global messages that are not passed to components
 	switch msg := msg.(type) {
 	case popViewMsg:
-		if len(m.componentStack) > 1 {
-			m.componentStack = m.componentStack[:len(m.componentStack)-1]
+		if m.stack.Len() > 1 {
+			m.stack.Pop()
```

Comment:
Let's get rid of this `Len()` check, let's make `Pop()` maintain the >1 invariant. Then we can get rid of the `Len()` helper too.